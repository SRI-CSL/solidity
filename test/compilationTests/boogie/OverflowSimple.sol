pragma solidity ^0.4.24;

/**
 * @notice invariant x == y
 */
contract Overflow {
 
    uint8 x;
    uint8 y;

    function inc_overflow() public {
        x = x + 1;
        y = y + 1;
    }

    function inc_no_overflow() public {
        x = x + 1;
        require(x > y);
        y = y + 1;
        // No require needed for y, the one for x and the invariant is sufficient
    }
}