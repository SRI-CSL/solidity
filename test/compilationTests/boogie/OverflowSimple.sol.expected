^Boogie program verifier finished with 5 verified, 5 errors$